### PR TITLE
(#132) list support

### PR DESCRIPTION
**Pull request by dannypsnl**

### CODE REVIEW

File: core/list/list.c [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+  struct List *next = head;
+  int index = 0;
+  while (index < size) {
+    next = push(next, *(values + index));
+    index++;
+  }
+  return head;
+}
+
+uint64_t list_length(struct List *list) {
```
Comment (LittleYenMin) [inline core/list/list.c:32]: Why don't you maintain a pointer of the length variable, and needs to loop the list?

Comment (LittleYenMin) [reply core/list/list.c:32]: Or it's an original package provided by C?

Comment (dannypsnl) [reply core/list/list.c:32]: yes

Comment (dannypsnl) [reply core/list/list.c:32]: Basically, manage the length cache, for now, is too complicated, so just need to sure functionality is correct.

**Review by LittleYenMin — APPROVED** [review]

**Pull request merged**